Modules/_posixsubprocess.c

```diff
@@ -1002,7 +994,7 @@ subprocess_fork_exec(PyObject *module, PyObject *args)
     /* Use vfork() only if it's safe. See the comment above child_exec(). */
     sigset_t old_sigs;
     if (preexec_fn == Py_None && allow_vfork &&
-        !call_setuid && !call_setgid && !call_setgroups) {
+        uid == (uid_t)-1 && gid == (gid_t)-1 && groups_list == Py_None) {
```

Comment:
for consistency, use `num_groups < 0` here and below instead of the groups_list Py_None check.